src/stormpath.social-login.js

```diff
@@ -0,0 +1,197 @@
+   */
+  SocialLoginService.prototype.getProviders = function getProviders() {
+    var providersPromise = this.providersPromise;
+    var initProviders = this.initProviders.bind(this);
+
+    if (providersPromise) {
+      return providersPromise.promise;
+    }
+
+    providersPromise = this.$q.defer();
+    this.providersPromise = providersPromise;
+
+    this.$http.get(this.STORMPATH_CONFIG.getUrl('OAUTH_PROVIDERS_ENDPOINT')).then(function(response) {
+      var providers = response.data;
+
+      initProviders(providers);
+
+      providersPromise.resolve(providers);
+    }).catch(providersPromise.reject);
+
+    return providersPromise.promise;
+  };
+
+  angular.module('stormpath.socialLogin', ['stormpath.CONFIG'])
+
+  /**
+   * @ngdoc object
+   *
+   * @name stormpath.socialLoginService.$socialLoginProvider
+   *
+   * @description
+   *
+   * Provides the {@link stormpath.socialLoginService.$socialLogin $socialLogin} service.
+   *
+   * Currently, this provider does not have any configuration methods.
+   */
+  .config(['$injector', 'STORMPATH_CONFIG', function $socialLoginProvider($injector, STORMPATH_CONFIG) {
+    /**
+     * @ngdoc object
+     *
+     * @name stormpath.socialLoginService.$socialLogin
+     *
+     * @description
+     *
+     * The social login service provides methods for letting users logging in with Facebook, Google, etc.
+     */
+    var socialLoginFactory = ['$http', '$q', '$injector', function socialLoginFactory($http, $q, $injector) {
+      return new SocialLoginService(STORMPATH_CONFIG, $injector, $http, $q);
+    }];
+
+    $injector.get('$provide').factory(STORMPATH_CONFIG.SOCIAL_LOGIN_SERVICE_NAME, socialLoginFactory);
+  }])
+
+  /**
+   * @ngdoc object
+   *
+   * @name stormpath.socialLoginService.$spJsLoader
+   *
+   * @description
+   *
+   * The `$spJsLoader` provides a method for loading scripts during runtime.
+   * Used by the social provider services to load their SDKs.
+   */
+  .factory('$spJsLoader', ['$q', function($q) {
+    return {
+      load: function load(id, src, innerHTML) {
+        var deferred = $q.defer();
+        var firstJsElement = document.getElementsByTagName('script')[0];
+        var jsElement = document.createElement('script');
+
+        if (document.getElementById(id)) {
+          deferred.resolve();
+        } else {
+          jsElement.id = id;
+          jsElement.src = src;
+          jsElement.innerHTML = innerHTML;
+          jsElement.onload = deferred.resolve;
+
+          firstJsElement.parentNode.insertBefore(jsElement, firstJsElement);
+        }
+
+        return deferred.promise;
+      }
+    };
+  }])
+
+  /**
+   * @ngdoc directive
+   *
+   * @name stormpath.spSocialLogin:spSocialLogin
+   *
+   * @description
+   *
+   * Add this directive to a button or link in order to authenticate using an OAuth provider. The value should be an OAuth provider id such as google or facebook.
```

Comment:
Ping @omgitstom - I want to make sure we're using the correct language to refer to the social providers.  Should we be calling it an OAuth provider, social provider, or directory provider?